The `log_softmax` operator fails accuracy tests sometimes on mthreads backend

### Describe the bug

The error context:

```json
{
  "tests/test_log_softmax.py::test_log_softmax[1-dtype0-shape2]": {
    "params": {
      "dim": 1,
      "dtype": "torch.float16",
      "shape": [
        200,
        40999,
        3
      ]
    },
    "result": "failed",
    "opname": [
      "log_softmax"
    ],
    "reason": "AttributeError: module 'torch._C' has no attribute '_cuda_getDevice'"
  }
}
```

3 out of 9 tests failed:

```
tests/test_log_softmax.py::test_log_softmax[1-dtype0-shape0] PASSED
tests/test_log_softmax.py::test_log_softmax[1-dtype0-shape1] PASSED
tests/test_log_softmax.py::test_log_softmax[1-dtype0-shape2] FAILED
tests/test_log_softmax.py::test_log_softmax[1-dtype1-shape0] PASSED
tests/test_log_softmax.py::test_log_softmax[1-dtype1-shape1] PASSED
tests/test_log_softmax.py::test_log_softmax[1-dtype1-shape2] FAILED
tests/test_log_softmax.py::test_log_softmax[1-dtype2-shape0] PASSED
tests/test_log_softmax.py::test_log_softmax[1-dtype2-shape1] PASSED
tests/test_log_softmax.py::test_log_softmax[1-dtype2-shape2] FAILED
```

The detailed log:

```
________________________________________________________________________ test_log_softmax[1-dtype0-shape2] ________________________________________________________________________

shape = (200, 40999, 3), dtype = torch.float16, dim = 1

    @pytest.mark.log_softmax
    @pytest.mark.parametrize("shape", utils.REDUCTION_SHAPES)
    @pytest.mark.parametrize("dtype", FLOAT_DTYPES)
    @pytest.mark.parametrize("dim", DIM_LIST)
    def test_log_softmax(shape, dtype, dim):
        if flag_gems.vendor_name == "cambricon":
            torch.manual_seed(42)
            torch.mlu.manual_seed_all(42)
    
        inp = torch.randn(shape, dtype=dtype, device=flag_gems.device)
        ref_inp = utils.to_reference(inp, True)
    
        ref_out = torch.nn.functional.log_softmax(ref_inp, dim=dim)
        with flag_gems.use_gems():
>           res_out = torch.nn.functional.log_softmax(inp, dim=dim)

tests/test_log_softmax.py:40: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
.venv/lib/python3.10/site-packages/torch/nn/functional.py:2248: in log_softmax
    ret = input.log_softmax(dim)
src/flag_gems/runtime/backend/_mthreads/ops/log_softmax.py:343: in log_softmax
    log_softmax_kernel_non_inner[grid](
.venv/lib/python3.10/site-packages/triton/runtime/jit.py:362: in <lambda>
    return lambda *args, **kwargs: self.run(grid=grid, warmup=False, *args, **kwargs)
src/flag_gems/utils/libentry.py:727: in run
    kernel = self.fn.run(*args, **kwargs)
.venv/lib/python3.10/site-packages/triton/runtime/autotuner.py:347: in run
    kwargs[v] = heur({**dict(zip(self.arg_names, args)), **kwargs})
src/flag_gems/runtime/backend/_nvidia/heuristics_config_utils.py:221: in softmax_heur_tile_k
    torch.cuda.current_device()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

    def current_device() -> int:
        r"""Return the index of a currently selected device."""
        _lazy_init()
>       return torch._C._cuda_getDevice()
E       AttributeError: module 'torch._C' has no attribute '_cuda_getDevice'

.venv/lib/python3.10/site-packages/torch/cuda/__init__.py:1027: AttributeError
```

--- a/src/flag_gems/runtime/backend/__init__.py
+++ b/src/flag_gems/runtime/backend/__init__.py
@@ -398,12 +398,9 @@ def get_unused_ops(vendor_name=None):
 
 def get_heuristic_config(vendor_name=None):
     config_name = "heuristics_config_utils"
+    vendor_name = vendor_name or "nvidia"
     mod_name = f"_{vendor_name}.{config_name}"
-    try:
-        _state.heuristic_config_module = importlib.import_module(mod_name)
-    except Exception:
-        mod_name = f"_nvidia.{config_name}"
-        _state.heuristic_config_module = importlib.import_module(mod_name)
+    _state.heuristic_config_module = importlib.import_module(mod_name)
     return getattr(_state.heuristic_config_module, "HEURISTICS_CONFIGS", None)
 
 

--- a/tests/test_backend_heuristics.py
+++ b/tests/test_backend_heuristics.py
@@ -1,0 +1,28 @@
+import pytest
+
+from flag_gems.runtime import backend
+
+
+def test_non_nvidia_heuristic_import_failure_does_not_fallback(monkeypatch):
+    imported_modules = []
+    import_module = backend.importlib.import_module
+
+    def fake_import_module(module_name):
+        imported_modules.append(module_name)
+        if module_name == "_mthreads.heuristics_config_utils":
+            raise ModuleNotFoundError("simulated mthreads heuristic import failure")
+        return import_module(module_name)
+
+    monkeypatch.setattr(backend.importlib, "import_module", fake_import_module)
+
+    with pytest.raises(ModuleNotFoundError, match="mthreads heuristic"):
+        backend.get_heuristic_config("mthreads")
+
+    assert "_nvidia.heuristics_config_utils" not in imported_modules
+
+
+def test_mthreads_softmax_non_inner_uses_mthreads_heuristic():
+    configs = backend.get_heuristic_config("mthreads")
+    tile_k = configs["softmax_non_inner"]["TILE_K"]
+
+    assert tile_k.__module__ == "_mthreads.heuristics_config_utils"

